internal/pipeline/io.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Package pipeline provides struct definitions for Pipeline and utility
+// methods to read and write a pipeline resource.
+package pipeline
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+const (
+	kptAPIVersion string = "kpt.dev/v1alpha1"
+	pipelineKind  string = "Pipeline"
+	defaultName   string = "pipeline"
+)
+
+var defaultSources []string = []string{"./*"}
+
+// newEmptyPipeline returns a new empty Pipeline
+func newEmptyPipeline() *Pipeline {
+	return &Pipeline{}
+}
+
+// NewPipeline returns a pointer to a new default Pipeline.
+// The default Pipeline should be:
+// apiVersion: kpt.dev/v1alpha1
+// kind: Pipeline
+// metadata:
+//   name: pipeline
+// sources:
+//   - '.*'
+func NewPipeline() *Pipeline {
```

Comment:
It's kind of unorthodox to have the different constructors be in a different file, any reason not do have these in types.go? I think `io.go` is bit of a misnomer anyways.

Reply:
Will merge these 2 files.